pasta_eln/GUI/projectTreeView.py

```diff
@@ -171,6 +172,61 @@ def tree2Clicked(self) -> None:
     return
 
 
+  def dragEnterEvent(self, event:QMouseEvent) -> None:
+    """
+    Overwrite default: what happens if you drag an item
+
+    Args:
+      event (QMouseEvent): event
+    """
+    event.acceptProposedAction()
+    return
+
+
+  def dropEvent(self, event:QMouseEvent) -> None:
+    """
+    Overwrite default: what happens at end of drag&drop
+
+    Args:
+      event (QMouseEvent): event
+    """
+    if event.mimeData().hasUrls():
+      item = self.model().itemFromIndex(self.indexAt(event.pos()))
+      if item.data()['docType'][0][0]!='x':
+        showMessage(self, 'Error', 'You cannot drop files on non folders.')
+        return
+      # create a list of all files
+      files, folders = [], []
+      for url in event.mimeData().urls():
+        path = url.toLocalFile()
+        if Path(path).is_file():
+          files.append(path)
+        else:
+          files +=   list(Path(path).rglob("*"))
+          folders += [x[0] for x in os.walk(path)]
+      docID = item.data()['hierStack'].split('/')[-1]
+      doc = self.comm.backend.db.getDoc(docID)
+      targetFolder = Path(self.comm.backend.cwd/doc['-branch'][0]['path'])
+      commonBase   = os.path.commonpath(folders+[str(i) for i in files])
+      # create folders and copy files
+      for folder in folders:
+        (targetFolder/(Path(folder).relative_to(commonBase))).mkdir(parents=True, exist_ok=True)
+      for fileStr in files:
+        file = Path(fileStr)
+        if file.is_file():
+          shutil.copy(file, targetFolder/(file.relative_to(commonBase)))
+      # scan
+      self.comm.backend.scanProject(self.comm.progressBar, docID, str(targetFolder) )
+      self.comm.changeProject.emit(item.data()['hierStack'].split('/')[0],'')
+      showMessage(self, 'Information','Drag & drop is finished')
+      event.ignore()
+    elif 'application/x-qstandarditemmodeldatalist' in event.mimeData().formats():
+      super().dropEvent(event)
+    else:
+      print('Drop unknown data', event.mimeData().formats())
```

Comment:
use log.warning! or log.error if you need to see the message in console too..!!

Reply:
You are correct